hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -317,10 +317,16 @@ private void handleFullBuffer() throws IOException {
       if (!commitWatcher.getFutureMap().isEmpty()) {
         waitOnFlushFutures();
       }
-    } catch (InterruptedException | ExecutionException e) {
+    } catch (ExecutionException e) {
       setIoException(e);
       adjustBuffersOnException();
       throw getIoException();
+    } catch (InterruptedException ex) {
+      LOG.error("Command execution was interrupted.");
+      Thread.currentThread().interrupt();
+      setIoException(ex);
+      adjustBuffersOnException();
```

Comment:
Same code is in multiple places, can we have short utility to do all these in that. So, this will avoid code smell in sonar.

Reply:
@bharatviswa504  I have tried to move the duplicate lines to separate method and also perform other cleanups. 
At the very least, the thread interrupt must be done in the catch block and not in a separate method being called from the catch block. So, the thread.interrupt() is the only line that cannot be avoided from being duplicated.